web3/manager.py

```diff
@@ -48,7 +48,10 @@ def middlewares(self, value):
     def _generate_request_functions(self):
         self._wrapped_provider_request_functions = {
             index: combine_middlewares(
-                middlewares=self._middlewares,
+                middlewares=tuple(itertools.chain(
+                    self._middlewares,
+                    provider.get_middlewares(),
+                )),
```

Comment:
perhaps just `middlewares=self.middlewares + provider.get_middlewares(),` ?

Reply:
A fair simplification.  I get a little carried away with my *functional* approach at times and need these kinds of reminders.